File: app/src/main/java/com/popalay/yocard/data/models/Holder.java
package com.popalay.yocard.data.models;

import com.github.nitrico.lastadapter.StableId;

import io.realm.RealmObject;
import io.realm.annotations.PrimaryKey;

public class Holder extends RealmObject implements StableId {

    public static final String ID = "id";
    public static final String NAME = "name";

    @PrimaryKey private long id;

    private String name;

    private int cardsCount;

    public Holder() {
    }

    public Holder(String name) {
        this.name = name;
    }

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public int getCardsCount() {
        return cardsCount;
    }

    public void setCardsCount(int cardsCount) {
        this.cardsCount = cardsCount;
    }

    @Override
    public long getStableId() {
        return id;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (o == null || getClass() != o.getClass()) {
            return false;
        }

        final Holder holder = (Holder) o;

        if (id != holder.id) {
            return false;
        }
        if (cardsCount != holder.cardsCount) {
            return false;
        }
        return name != null ? name.equals(holder.name) : holder.name == null;

    }

    @Override
    public int hashCode() {
        int result = (int) (id ^ (id >>> 32));
        result = 31 * result + (name != null ? name.hashCode() : 0);
        result = 31 * result + cardsCount;
        return result;
    }
}

File: app/src/main/java/com/popalay/cardme/data/models/Holder.kt
package com.popalay.cardme.data.models

import com.github.nitrico.lastadapter.StableId
import io.realm.RealmObject
import io.realm.annotations.PrimaryKey

open class Holder(
        @PrimaryKey open var id: String? = null,
        open var name: String = "",
        open var cardsCount: Int = 0,
        open var debtCount: Int = 0
) : RealmObject(), StableId {

    companion object {
        const val ID = "id"
        const val NAME = "name"
        const val CARDS_COUNT = "cardsCount"
        const val DEBTS_COUNT = "debtCount"
    }

    override val stableId: Long
        get() = id?.hashCode()?.toLong() ?: 0L

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other?.javaClass != javaClass) return false
        other as Holder
        if (id != other.id) return false
        if (name != other.name) return false
        if (cardsCount != other.cardsCount) return false
        if (debtCount != other.debtCount) return false
        return true
    }

    override fun hashCode(): Int {
        var result = id?.hashCode() ?: 0
        result = 31 * result + name.hashCode()
        result = 31 * result + cardsCount
        result = 31 * result + debtCount
        return result
    }
}
